### PR TITLE
Build a Rotten Tomatoes Clone with GraphQL and Auth0

### PR DESCRIPTION
**Pull request by unicodeveloper**

### CODE REVIEW

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+
+// For Authentication
+networkInterface.use([{
+  applyMiddleware (req, next) {
+    if (!req.options.headers) {
+      req.options.headers = {}
+    }
+    // get the authentication token from local storage if it exists
+    if (localStorage.getItem('id_token')) {
+      req.options.headers.authorization = `Bearer ${localStorage.getItem('id_token')}`
```
Comment (sebadoom) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: Why are you putting the id_token in the headers?

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown [hunk 2]
```diff
@@ -0,0 +1,957 @@
...
+  return date;
+}
+
+function isTokenExpired(token) {
+  const expirationDate = getTokenExpirationDate(token);
+  return expirationDate < new Date();
+}
+```
+
+In the code above, we are using an hosted version of Auth0 Lock in the `login` method and passed in our credentials.
```
Comment (sebadoom) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: Do not mention Auth0 Lock here, it is confusing. Instead, say "hosted login page".

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown [hunk 3]
```diff
@@ -0,0 +1,957 @@
...
+![Configure Auth Provider](https://cdn.auth0.com/blog/rottentomatoes/configure-auth-provider.png)
+
+Click on Integrations and enable Auth0
+
+![Click on Integrations](https://cdn.auth0.com/blog/rottentomatoes/activate.png)
+
+![Integrations Dialog Box](https://cdn.auth0.com/blog/rottentomatoes/enable-auth0.png)
+
+Copy your Auth0 Client Credentials and paste it here.
+![Add Client Details](https://cdn.auth0.com/blog/rottentomatoes/add-client-details.png)
```
Comment (sebadoom) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: You said above the API should use RS256, yet here the screenshot requests the client secret (HS256). This is confusing.

Comment (sebadoom) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: Also, it is not clear at what point you configured the GraphQL client to send the access_token to the GraphCool server. Where is that?

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown [hunk 4]
```diff
@@ -0,0 +1,957 @@
...
+![Insufficient Permission](https://cdn.auth0.com/blog/rottentomatoes/insufficient-permission.png)
+
+GraphQL has now made it impossible for just any user to create a movie. You have to be authenticated to have the privilege! Whoop! Whoop!
+
+
+## Conclusion
+
+In this tutorial, we covered how easy it is to build a product with a backend like GraphCool and add authentication to it easily using Auth0.
+
+In addition, Auth0 can help secure yourapps with more than just username-password authentication. It provides features like [multifactor auth](https://auth0.com/docs/multifactor-authentication), [anomaly detection](https://auth0.com/docs/anomaly-detection), [enterprise federation](https://auth0.com/docs/identityproviders), [single sign on (SSO)](https://auth0.com/docs/sso), and more. [Sign up](javascript:signup\(\)) today so you can focus on building features unique to your app.
```
Comment (sebadoom) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: yourapps -> your apps

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown (hunk 3 above)
Comment (unicodeveloper) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: It is at the point where you send id token via the headers.

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown (hunk 1 above)
Comment (unicodeveloper) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: When making a request to the graph cool server via the Apollo client, this is the recommended way. Any particular concerns about this?

File: _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown (hunk 3 above)
Comment (unicodeveloper) [inline _posts/2017-06-30-rottentomatoes-clone-with-graphql-and-auth0.markdown]: Where did you see HS256? The screenshot shows RS256 @sebadoom

**Pull request merged**